aws_lambda_powertools/event_handler/openapi/compat.py

```diff
@@ -15,13 +15,27 @@
 from pydantic import BaseModel, create_model
 from pydantic.fields import FieldInfo
 
-from aws_lambda_powertools.event_handler.openapi.pydantic_loader import PYDANTIC_V2
 from aws_lambda_powertools.event_handler.openapi.types import (
     COMPONENT_REF_PREFIX,
     ModelNameMap,
     UnionType,
 )
 
+from pydantic import TypeAdapter, ValidationError
+from pydantic._internal._typing_extra import eval_type_lenient
```

Comment:
is there another more stable way? 

`_internal` is private so this **might** cause breaking changes. If there's no other way, we need to make a comment of the risk